File: selfstudy/golang/words_reverse.go
/*
task 298:
Words - Reverse
You need to reverse the order of word lengths
*/

package main

import (
	"fmt"
	"reflect"
)

func Reverse_orderWords(lengths []int) []int {
	res := make([]int, len(lengths))
	for i, v := range lengths {
		res[len(lengths)-1-i] = v
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Reverse_orderWords([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Reverse_orderWords([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Reverse_orderWords([]int{1, 2, 3})},
		{[]int{100, 200}, Reverse_orderWords([]int{100, 200})},
		{[]int{}, Reverse_orderWords([]int{})},
	}
	
	for idx, t := range tests {
		res := Reverse_orderWords(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done

File: selfstudy/js/words_reverse.js
/*
task 98:
Words - Reverse
You need to reverse the order of word lengths
*/

function reverse_order_words(lengths) {
  return [...lengths].reverse();
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: reverse_order_words([10, 20, 30]) },
  { input: [50, 60, 70], expected: reverse_order_words([50, 60, 70]) },
  { input: [1, 2, 3], expected: reverse_order_words([1, 2, 3]) },
  { input: [100, 200], expected: reverse_order_words([100, 200]) },
  { input: [], expected: reverse_order_words([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('reverse_order_words', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(reverse_order_words(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, reverse_order_words(t.input) === t.expected || JSON.stringify(reverse_order_words(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// finally works
